src/Tasks/Microsoft.NET.Testing.Tasks/GetTestsProject.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
+
+using Microsoft.Build.Framework;
+using Microsoft.DotNet.Tools.Test;
+
+namespace Microsoft.NET.Sdk.Testing.Tasks
+{
+    public class GetTestsProject : Build.Utilities.Task
+    {
+        [Required]
+
+        public ITaskItem TargetPath { get; set; }
+
+        [Required]
+
+        public ITaskItem GetTestsProjectPipeName { get; set; }
+
+        public override bool Execute()
+        {
+            try
+            {
+                Log.LogMessage(MessageImportance.Low, $"Target path: {TargetPath}");
+
+                NamedPipeClient dotnetTestPipeClient = new(GetTestsProjectPipeName.ItemSpec);
```

Comment:
Right now there are going to be quite a few instances of this pipe client created over the course of a build:

* one per outer build
* one per inner build (aka per-TFM)

We suggest instead that you use the MSBuild Engine's [RegisterTaskObject API](https://learn.microsoft.com/en-us/dotnet/api/microsoft.build.framework.ibuildengine4.registertaskobject?view=msbuild-17-netcore) to create one client per `GetTestsProjectPipeName` and re-use it. If you use this API and your type also implements `IDisposable`, then MSBuild will also make sure that it is disposed correctly.

Reply:
This is interesting thanks